docs/reference/platforms.rst

```diff
@@ -0,0 +1,218 @@
+=========
+
+Platforms
+---------
+
+The ``platforms`` keyword in a ``charmcraft.yaml`` determines where charms are
+built and where they run.
+
+For more information on how to build charms for specific bases and
+architectures, see the :doc:`Platforms how-to </howto/build-guides/platforms>`
+page.
+
+Standard notation
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on: [<arch>, <arch>]
+       build-for: [<arch>]
+
+The platform name is an arbitrary string that describes the platform. The
+recommended platform name is the ``build-for`` arch.
+
+See :ref:`build-on <reference-build-on>` and
+:ref:`build-for <reference-build-for>` for information on the ``build-on``
+and ``build-for`` keywords.
+
+.. _reference-platforms-shorthand:
+
+Shorthand notation
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <arch>:
+
+Shorthand notation is a simple way to describe charms that build on and build
+for the same architecture. Shorthand notation requires the platform name to be
+a :ref:`valid debian architecture <reference-architectures>`.
+
+.. _reference-platforms-multi-base:
+
+Multi-base notation
+~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on: [<base>:<arch>, <base>:<arch>]
+       build-for: [<base>:<arch>]
+
+Multi-base charms define a :ref:`base <reference-bases>` in each platform entry
+instead of the top-level ``base`` and ``build-base`` keywords. Within each
+platform entry, the base must be the same for all ``build-on`` and
+``build-for`` entries.
+
+The recommended platform name is ``<distribution>-<series>-<build-for-arch>``
+(for example, ``ubuntu-24.04-riscv64``).
+
+.. _reference-platforms-multi-base-shorthand:
+
+Multi-base shorthand notation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <base>:<arch>:
+
+Multi-base charms can also use shorthand notation when the platform builds on
+and builds for the same architecture. The platform name must be a
+:ref:`valid base <reference-bases>` and
+:ref:`debian architecture <reference-architectures>` formatted as
+``<base>:<arch>``.
+
+.. _reference-build-on:
+
+``build-on``
+~~~~~~~~~~~~
+
+``build-on`` is a list of architectures and optional bases that describes the
+environments where the platform can build. Each entry is formatted as
+``[<base>:]<arch>``.
+
+.. _reference-build-for:
+
+``build-for``
+~~~~~~~~~~~~~
+
+``build-for`` is a single-element list containing an architecture and optional
+base that describes the environment where the resulting charm can run. The
+entry is formatted as ``[<base>:]<arch>``.
+
+``build-for: [all]`` is a special keyword to denote an architecture-independent
+charm.
+
+.. _reference-architectures:
+
+Architectures
+-------------
+
+Charmcraft uses `Debian's naming convention`_ for architectures.
+
+The following architectures are supported:
+
+* amd64
+* arm64
+* armhf
+* i386
+* ppc64el
+* riscv64
+* s390x
```

Comment:
It might be worth mapping these to human-friendly names in a table. I compiled the list in the [Snapcraft system reqs](https://github.com/canonical/snapcraft/blob/685a8d61a6f378179334a12ab868ddc414e7bd41/docs/reference/system-requirements.rst?plain=1#L14).

Reply:
Turned it into a table, removed i386 (which Charmcraft doesn't support), and added a note about Juju support. Sadly there doesn't seem to be a Juju architecture support doc I can reference.